### PR TITLE
docs(guide): convert /tutorials/text-search to rst

### PR DESCRIPTION
**Pull request by daprahamian**
Fixes NODE-2210

## Description

**What changed?**

**Are there any files to ignore?**

### CODE REVIEW

File: docs/guide/tutorials/text-search.txt [hunk 1]
```diff
@@ -1,3 +1,75 @@
...
+       console.log(results);
+     } finally {
+       await client.close();
+     }
+   });
+
+The following example assumes that a database called ``test`` has a
+collection called ``restaurants``\ , with a text index on the ``name`` field.
+
+.. raw::html
```
Comment (daprahamian) [inline docs/guide/tutorials/text-search.txt]: Originally had a line here saying:

> A [sample dataset](https://docs.mongodb.org/getting-started/node/import-data/) is available for download.

But that just redirects to http://mongodb.github.io/node-mongodb-native/2.2/quick-start/quick-start/ which has no sample dataset.  Is there a proper place to get this "dataset"?

File: docs/guide/tutorials/text-search.txt [hunk 2]
```diff
@@ -1,3 +1,75 @@
...
+   const url = 'mongodb://localhost:27017';
+   
+   // Database Name
+   const dbName = 'myproject';
+   
+   // Create a new MongoClient
+   const client = new MongoClient(url);
+   
+   // Use connect method to connect to the Server
+   client.connect().then(async function() {
```
Comment (mongomoe) [inline docs/guide/tutorials/text-search.txt]: ` client.connect().then(async function() {` should be es6 arrow fn : `client.connect().then(async () => {`

File: docs/guide/tutorials/text-search.txt [hunk 3]
```diff
@@ -1,3 +1,75 @@
...
```
Comment (mongomoe) [inline docs/guide/tutorials/text-search.txt]: ` client.connect().then(async function() {` should be es6 arrow fn : `client.connect().then(async () => {`

**Comment by mongomoe** [comment]
This might be beyond the scope of the initial migration/conversion to rst, but this doc (and some of the others) assume the reader has data in their local mongodb database (and without a transparent explanation on how to import it or where to get the data). This might confuse readers. Perhaps we can solve this issue by:

1. Uploading the dataset(s) to s3 (or github or some other resource to serve static files) and then give instructions on how to import it locally
or
2. Upload the dataset(s) to an atlas cluster and give read access to the cluster, then use the remote URL for these docs. 

Solution 2 is probably the most simple for readers in my opinion. This isn't required for LGTM for this specific ticket, it's just some thoughts on the subject.

**Comment by daprahamian** [comment]
@mongomoe yeah, the original version of the guide included a dataset, but that link appears to be lost. You can see the that [here](http://mongodb.github.io/node-mongodb-native/2.2/tutorials/text-search/):

> A [sample dataset](https://docs.mongodb.org/getting-started/node/import-data/) is available for download.

That link just reroutes to the quickstart page.

I'm fine with whatever solution you want. Do you want to do that now, or should we make a ticket for doing that?

**Comment by daprahamian** [comment]
@mongomoe I've addressed all the feedback points outside of the point of providing a dataset.

**Pull request merged**